exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs

```diff
@@ -50,43 +50,54 @@ impl JavaServiceRuntime {
         }
     }
 
-    /// Returns internal service proxy.
-    pub fn service_proxy(&self) -> ServiceProxy {
-        self.service_proxy.clone()
-    }
-
-    /// Initializes JVM with provided configuration.
-    ///
-    /// # Panics
-    ///
-    /// - If user specified invalid additional JVM parameters.
-    fn create_java_vm(jvm_config: JvmConfig, ejb_config: EjbConfig) -> JavaVM {
+    /// Builds arguments for JVM initialization.
+    fn build_jvm_arguments(jvm_config: JvmConfig, ejb_config: EjbConfig) -> Result<InitArgs> {
         let mut args_builder = jni::InitArgsBuilder::new().version(jni::JNIVersion::V8);
 
+        // Prepend extra user arguments
         for param in &jvm_config.args_prepend {
```

Comment:
Isn't this loop, duplicated below, extractable into `addUserArguments(args_builder, options)`, as was suggested?